pkg/generate/app/pipeline.go

```diff
@@ -311,6 +311,30 @@ func objectMetaData(raw interface{}) (runtime.Object, *kapi.ObjectMeta, error) {
 	return obj, meta, nil
 }
 
+type acceptBuildConfigs struct {
+	typer runtime.ObjectTyper
+}
+
+// Accept accepts just BuildConfigs
+func (a *acceptBuildConfigs) Accept(from interface{}) bool {
+	obj, _, err := objectMetaData(from)
+	if err != nil {
+		return false
+	}
+	_, kind, err := a.typer.ObjectVersionAndKind(obj)
+	if err != nil {
+		return false
+	}
+	return kind == "BuildConfig"
```

Comment:
Always end an output image stream from new-build, unless we support an argument that lets you push to an external Docker repo (nice to have, can be post 3.0)

----- Original Message -----
> > @@ -311,6 +311,30 @@ func objectMetaData(raw interface{}) (runtime.Object,
> > *kapi.ObjectMeta, error) {
> >  	return obj, meta, nil
> >  }
> >  
> > +type acceptBuildConfigs struct {
> > +	typer runtime.ObjectTyper
> > +}
> > +
> > +// Accept accepts just BuildConfigs
> > +func (a *acceptBuildConfigs) Accept(from interface{}) bool {
> > +	obj, _, err := objectMetaData(from)
> > +	if err != nil {
> > +		return false
> > +	}
> > +	_, kind, err := a.typer.ObjectVersionAndKind(obj)
> > +	if err != nil {
> > +		return false
> > +	}
> > +	return kind == "BuildConfig"
> 
> What about the output image stream? or the input image stream if creating
> from a docker repo?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2747/files#r31661250